tests/src/core/testqgsvectordataprovider.cpp

```diff
@@ -224,6 +242,75 @@ void TestQgsVectorDataProvider::featureAtId()
   QVERIFY( !feature.isValid() );
 }
 
+void TestQgsVectorDataProvider::sourceExtent()
+{
+  // 2d data
+  QgsVectorDataProvider *prLines = vlayerLines->dataProvider();
+
+  QGSCOMPARENEAR( prLines->sourceExtent().xMinimum(), -117.623, 0.001 );
+  QGSCOMPARENEAR( prLines->sourceExtent().xMaximum(), -82.3226, 0.001 );
+  QGSCOMPARENEAR( prLines->sourceExtent().yMinimum(), 23.2082, 0.001 );
+  QGSCOMPARENEAR( prLines->sourceExtent().yMaximum(), 46.1829, 0.001 );
+
+  QGSCOMPARENEAR( prLines->sourceExtent3D().xMinimum(), -117.623, 0.001 );
+  QGSCOMPARENEAR( prLines->sourceExtent3D().xMaximum(), -82.3226, 0.001 );
+  QGSCOMPARENEAR( prLines->sourceExtent3D().yMinimum(), 23.2082, 0.001 );
+  QGSCOMPARENEAR( prLines->sourceExtent3D().yMaximum(), 46.1829, 0.001 );
+  QVERIFY( std::isnan( prLines->sourceExtent3D().zMinimum() ) );
+  QVERIFY( std::isnan( prLines->sourceExtent3D().zMaximum() ) );
+
+
+  QgsVectorDataProvider *prPoints = vlayerPoints->dataProvider();
+
+  QGSCOMPARENEAR( prPoints->sourceExtent().xMinimum(), -118.889, 0.001 );
+  QGSCOMPARENEAR( prPoints->sourceExtent().xMaximum(), -83.3333, 0.001 );
+  QGSCOMPARENEAR( prPoints->sourceExtent().yMinimum(), 22.8002, 0.001 );
+  QGSCOMPARENEAR( prPoints->sourceExtent().yMaximum(), 46.872, 0.001 );
+
+  QGSCOMPARENEAR( prPoints->sourceExtent3D().xMinimum(), -118.889, 0.001 );
+  QGSCOMPARENEAR( prPoints->sourceExtent3D().xMaximum(), -83.3333, 0.001 );
+  QGSCOMPARENEAR( prPoints->sourceExtent3D().yMinimum(), 22.8002, 0.001 );
+  QGSCOMPARENEAR( prPoints->sourceExtent3D().yMaximum(), 46.872, 0.001 );
+  QVERIFY( std::isnan( prPoints->sourceExtent3D().zMinimum() ) );
+  QVERIFY( std::isnan( prPoints->sourceExtent3D().zMaximum() ) );
+
+  // 3d data
+  QgsVectorDataProvider *prLines3D = vlayerLines3D->dataProvider();
+
+  QGSCOMPARENEAR( prLines3D->sourceExtent().xMinimum(), 0.0, 0.001 );
+  QGSCOMPARENEAR( prLines3D->sourceExtent().xMaximum(), 322355.7101, 0.01 );
+  QGSCOMPARENEAR( prLines3D->sourceExtent().yMinimum(), 0.0, 0.001 );
+  QGSCOMPARENEAR( prLines3D->sourceExtent().yMaximum(), 129791.2638, 0.01 );
+
+  QGSCOMPARENEAR( prLines3D->sourceExtent3D().xMinimum(), 0.0, 0.001 );
+  QGSCOMPARENEAR( prLines3D->sourceExtent3D().xMaximum(), 322355.7101, 0.01 );
+  QGSCOMPARENEAR( prLines3D->sourceExtent3D().yMinimum(), 0.0, 0.001 );
+  QGSCOMPARENEAR( prLines3D->sourceExtent3D().yMaximum(), 129791.2638, 0.01 );
+  // still nan as provider implementation is not done
+  QVERIFY( std::isnan( prLines3D->sourceExtent3D().zMinimum() ) );
+  QVERIFY( std::isnan( prLines3D->sourceExtent3D().zMaximum() ) );
+//  QGSCOMPARENEAR( prLines3D->sourceExtent3D().zMinimum(), -5.00, 0.001 );
```

Comment:
Remove commented parts or as I know, it's for a next MR/release, add a TODO and explain why.

Reply:
fixed!